src/Raven.Server/Documents/Sharding/Handlers/Processors/Subscriptions/ShardedSubscriptionsHandlerProcessorForDropSubscriptionConnection.cs

```diff
@@ -16,40 +12,19 @@ public ShardedSubscriptionsHandlerProcessorForDropSubscriptionConnection([NotNul
         {
         }
 
-        protected override async ValueTask DropSubscriptionAsync(long? subscriptionId, string subscriptionName, string workerId)
+        protected override bool DropSingleSubscriptionConnection(SubscriptionState subscription, string workerId)
         {
-            if (subscriptionId.HasValue == false && string.IsNullOrEmpty(subscriptionName))
-                throw new ArgumentException("Subscription drop operation must get either subscription name or subscription task id");
+            bool result = RequestHandler.DatabaseContext.SubscriptionsStorage.DropSubscriptionConnections(subscription.SubscriptionId,
+                    new SubscriptionClosedException(
+                        $"Dropped by API request (request ip:{HttpContext.Connection.RemoteIpAddress}, cert:{HttpContext.Connection.ClientCertificate?.Thumbprint})",
+                        canReconnect: false));
 
-            if (subscriptionId.HasValue == false)
-            {
-                using (ContextPool.AllocateOperationContext(out TransactionOperationContext context))
-                using (context.OpenReadTransaction())
-                {
-                    subscriptionId = ServerStore.Cluster.Subscriptions.ReadSubscriptionStateByName(context, RequestHandler.DatabaseName, subscriptionName)
-                        .SubscriptionId;
-                }
-            }
-
-            if (string.IsNullOrEmpty(subscriptionName))
-            {
-                using (ContextPool.AllocateOperationContext(out TransactionOperationContext context))
-                using (context.OpenReadTransaction())
-                {
-                    subscriptionName = ServerStore.Cluster.Subscriptions.ReadSubscriptionStateById(context, RequestHandler.DatabaseName, subscriptionId.Value).SubscriptionName;
-                }
-            }
-
-            var op = new ShardedDropSubscriptionConnectionOperation(HttpContext, subscriptionName);
-            await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(op);
-            
-            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Stav, DevelopmentHelper.Severity.Normal, "RavenDB-19079 Make this identical to the normal EP");
-            if (RequestHandler.DatabaseContext.SubscriptionsStorage.Subscriptions.TryRemove(subscriptionId.Value, out var state))
-            {
-                state.Dispose();
-            }
+            return result;
+        }
 
-            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Stav, DevelopmentHelper.Severity.Normal, "RavenDB-19079 Handle status code");
+        protected override SubscriptionState GetRunningSubscriptionState(long? subscriptionId, string subscriptionName, TransactionOperationContext context)
+        {
+            return RequestHandler.DatabaseContext.SubscriptionsStorage.GetActiveSubscription(context, subscriptionId, subscriptionName);
```

Comment:
1. rebase :)
2. pass AbstractSubscriptionStorage (maybe actually I did that already - please rebase first) to the base class, then you will be able to remove this

Reply:
done